src/test/java/mate/academy/security/AuthenticationServiceImplTest.java

```diff
@@ -0,0 +1,124 @@
+package mate.academy.security;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.service.RoleService;
+import mate.academy.service.UserService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+class AuthenticationServiceImplTest {
+    private AuthenticationService authenticationService;
+    private UserService userService;
+    private RoleService roleService;
+    private PasswordEncoder passwordEncoder;
+    private User user;
+    private List<Role> roles;
+
+    @BeforeEach
+    void setUp() {
+        userService = Mockito.mock(UserService.class);
+        roleService = Mockito.mock(RoleService.class);
+        passwordEncoder = Mockito.mock(PasswordEncoder.class);
+        authenticationService = new AuthenticationServiceImpl(userService,
+                roleService, passwordEncoder);
+        user = new User();
+        user.setEmail("bob@i.ua");
+        user.setPassword("1234");
+        roles = new ArrayList<>();
+        roles.add(new Role(Role.RoleName.ADMIN));
+        roles.add(new Role(Role.RoleName.USER));
+    }
+
+    @Test
+    void register_OK() {
+        Mockito.when(roleService.getRoleByName(any())).thenReturn(roles.get(1));
+        user.setRoles(Set.of(roles.get(1)));
+        Mockito.when(userService.save(any())).thenReturn(user);
```

Comment:
Here and in the other places.
We should not use any() by default in our tests. Using any() is a potential mistake place. When you know exactly the argument that will be used inside your service method you should use it for mock objects.